src/Data/List/Relation/Binary/Sublist/Propositional/Properties.agda

```diff
@@ -158,3 +193,80 @@ IsWeakPushout {τ = τ} {σ = σ} rpo =
 ⊆-pushoutˡ-is-wpo {τ = y   ∷ʳ _}                = cong (y   ∷ʳ_) ⊆-pushoutˡ-is-wpo
 ⊆-pushoutˡ-is-wpo {τ = _   ∷  _} {σ = z   ∷ʳ _} = cong (z   ∷ʳ_) ⊆-pushoutˡ-is-wpo
 ⊆-pushoutˡ-is-wpo {τ = refl ∷ _} {σ = refl ∷ _} = cong (refl ∷_) ⊆-pushoutˡ-is-wpo
+
+------------------------------------------------------------------------
+-- Properties of disjointness
+
+-- From τ ⊎ σ = ρ, compute the difference τ′ such that τ = ⊆-trans τ′ ρ.
+
+DisjointUnion-diffˡ : ∀ {xs ys zs us : List A} {τ : xs ⊆ zs} {σ : ys ⊆ zs} {ρ : us ⊆ zs} →
+                      DisjointUnion τ σ ρ → ∃ λ (τ′ : xs ⊆ us) → ⊆-trans τ′ ρ ≡ τ
+DisjointUnion-diffˡ []         = []       , refl
+DisjointUnion-diffˡ (y   ∷ₙ d) = _        , cong (y  ∷ʳ_) (proj₂ (DisjointUnion-diffˡ d))
+DisjointUnion-diffˡ (x≈y ∷ₗ d) = refl ∷ _ , cong (x≈y ∷_) (proj₂ (DisjointUnion-diffˡ d))
+DisjointUnion-diffˡ (x≈y ∷ᵣ d) = _ ∷ʳ _   , cong (_  ∷ʳ_) (proj₂ (DisjointUnion-diffˡ d))
+
+-- From τ ⊎ σ = ρ, compute the difference σ′ such that σ = ⊆-trans σ′ ρ.
+
+DisjointUnion-diffʳ : ∀ {xs ys zs us : List A} {τ : xs ⊆ zs} {σ : ys ⊆ zs} {ρ : us ⊆ zs} →
+                      DisjointUnion τ σ ρ → ∃ λ (σ′ : ys ⊆ us) → ⊆-trans σ′ ρ ≡ σ
+DisjointUnion-diffʳ []         = []       , refl
+DisjointUnion-diffʳ (y   ∷ₙ d) = _        , cong (y  ∷ʳ_) (proj₂ (DisjointUnion-diffʳ d))
+DisjointUnion-diffʳ (x≈y ∷ᵣ d) = refl ∷ _ , cong (x≈y ∷_) (proj₂ (DisjointUnion-diffʳ d))
+DisjointUnion-diffʳ (x≈y ∷ₗ d) = _ ∷ʳ _   , cong (_  ∷ʳ_) (proj₂ (DisjointUnion-diffʳ d))
+
+------------------------------------------------------------------------
+-- A Union where the triangles commute is a
+-- Cospan in the slice category (_ ⊆ zs).
+
+record IsCospan {xs ys zs : List A} {τ₁ : xs ⊆ zs} {τ₂ : ys ⊆ zs} (u : UpperBound τ₁ τ₂) : Set a where
+  field
+    tri₁ : ⊆-trans (UpperBound.inj₁ u) (UpperBound.sub u) ≡ τ₁
+    tri₂ : ⊆-trans (UpperBound.inj₂ u) (UpperBound.sub u) ≡ τ₂
```

Comment:
These definitions should not be in `.Properties`. But as they rely on `⊆-trans` then I think all of this stuff needs to be moved out to it's own module. `Sublist.Heterogeneous/Setoid/Propositional.Disjoint` maybe?